### PR TITLE
Obsolete gateway extension 

### PR DESCRIPTION
**Pull request by HEskandari**

### CODE REVIEW

**Review by aleksandr-samila — APPROVED** [review]

File: src/NServiceBus.Gateway/Configuration/GatewayExtensions.cs [hunk 1]
```diff
@@ -12,6 +12,10 @@ public static class GatewayExtensions
         /// Allows the user to control how the gateway behaves.
         /// </summary>
         /// <param name="config">The <see cref="EndpointConfiguration"/> instance to apply the settings to.</param>
+        [ObsoleteEx(
+            TreatAsErrorFromVersion = "4",
+            RemoveInVersion = "5",
+            Message = "Gateway with no configuration is not supported. Use `Gateway(GatewayDeduplicationConfiguration)` instead.")]
```
Comment (mikeminutillo) [inline src/NServiceBus.Gateway/Configuration/GatewayExtensions.cs:18]: It is still supported in v7. Should this say something like "Provide deduplication configuration."?

**Review by mikeminutillo — CHANGES_REQUESTED** [review]

File: src/NServiceBus.Gateway/Configuration/GatewayExtensions.cs (hunk 1 above)
Comment (HEskandari) [reply src/NServiceBus.Gateway/Configuration/GatewayExtensions.cs:18]: @mikeminutillo how does this look? Don't know the best way to point to the overload.

```
ReplacementTypeOrMember = "EndpointConfiguration.Gateway(GatewayDeduplicationConfiguration)"
```

Comment (mikeminutillo) [reply src/NServiceBus.Gateway/Configuration/GatewayExtensions.cs:18]: I think that's probably perfect

Comment (HEskandari) [reply src/NServiceBus.Gateway/Configuration/GatewayExtensions.cs]: Great!

**Review by mikeminutillo — APPROVED** [review]

**Pull request merged**